presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1174,8 +1177,9 @@ public void testNumericTypeTranslation()
             throws Exception
     {
         List<NumericValues> translationChain = ImmutableList.of(
-                new NumericValues(C_DOUBLE, (value) -> doubleLiteral((double) value), -1.0 * Double.MAX_VALUE, -22.0, -44.555678, 23.0, 44.555678, Double.MAX_VALUE),
-                new NumericValues(C_DECIMAL_26_5, (value) -> decimalLiteral(Decimals.toString((Slice) value, 5)), longDecimal("-999999999999999999999.99999"), longDecimal("-22.00000"), longDecimal("-44.55568"), longDecimal("23.00000"), longDecimal("44.55567"), longDecimal("999999999999999999999.99999")),
+                new NumericValues(C_DOUBLE, (value) -> doubleLiteral((double) value), -1.0 * Double.MAX_VALUE, -22.0, -44.5556836, 23.0, 44.5556789, Double.MAX_VALUE),
+                new NumericValues(C_FLOAT, (value) -> floatLiteral(intBitsToFloat(((Long) value).intValue())), floatValue(-1.0f * Float.MAX_VALUE), floatValue(-22.0f), floatValue(-44.555687f), floatValue(23.0f), floatValue(44.555676f), floatValue(Float.MAX_VALUE)),
+                new NumericValues(C_DECIMAL_26_5, (value) -> decimalLiteral(Decimals.toString((Slice) value, 5)), longDecimal("-999999999999999999999.99999"), longDecimal("-22.00000"), longDecimal("-44.55569"), longDecimal("23.00000"), longDecimal("44.55567"), longDecimal("999999999999999999999.99999")),
```

Comment:
`DOUBLE` is floor casted to `FLOAT` and `DECIMAL`, and `FLOAT` is casted to `DECIMAL`. Everything is right.